domains/certificates/RegistryConnector/Worker/EventHandlers/MeasurementEventHandler.cs

```diff
@@ -0,0 +1,221 @@
+    }
+
+    private static bool ShouldEventBeProduced(CertificateIssuingContract contract, EnergyMeasuredIntegrationEvent energyMeasuredIntegrationEvent)
+    {
+        if (!contract.Contains(energyMeasuredIntegrationEvent.DateFrom, energyMeasuredIntegrationEvent.DateTo))
+            return false;
+
+        if (energyMeasuredIntegrationEvent.Quantity <= 0)
+            return false;
+
+        if (energyMeasuredIntegrationEvent.Quantity > uint.MaxValue)
+            return false;
+
+        if (energyMeasuredIntegrationEvent.Quality != MeasurementQuality.Measured)
+            return false;
+
+        return true;
+    }
+
+    private void BuildRoutingSlip(RoutingSlipBuilder builder,
+        Transaction transaction,
+        SecretCommitmentInfo commitment,
+        CertificateIssuingContract matchingContract,
+        uint walletDepositEndpointPosition)
+    {
+        var certificateId = Guid.Parse(transaction.Header.FederatedStreamId.StreamId.Value);
+
+        AddActivity<IssueToRegistryActivity, IssueToRegistryArguments>(builder, new IssueToRegistryArguments(transaction, certificateId));
+
+        AddActivity<WaitForCommittedTransactionActivity, WaitForCommittedTransactionArguments>(builder,
+            new WaitForCommittedTransactionArguments(transaction.ToShaId(), certificateId));
+
+        AddActivity<MarkAsIssuedActivity, MarkAsIssuedArguments>(builder,
+            new MarkAsIssuedArguments(certificateId, matchingContract.MeteringPointType));
+
+        var receiveRequest = new ReceiveRequest
+        {
+            CertificateId = transaction.Header.FederatedStreamId,
+            Quantity = commitment.Message,
+            RandomR = ByteString.CopyFrom(commitment.BlindingValue),
+            WalletDepositEndpointPublicKey = ByteString.CopyFrom(matchingContract.WalletPublicKey),
+            WalletDepositEndpointPosition = walletDepositEndpointPosition
+        };
+        AddActivity<SendToWalletActivity, SendToWalletArguments>(builder,
+            new SendToWalletArguments(matchingContract.WalletUrl, receiveRequest));
+
+        var issueCertificateFailedConsumerEndpoint = new Uri($"exchange:{endpointNameFormatter.Consumer<IssueCertificateNotCompletedConsumer>()}");
+        builder.AddSubscription(issueCertificateFailedConsumerEndpoint, RoutingSlipEvents.Terminated,
+            x => x.Send(new IssueCertificateTerminated
+            {
+                CertificateId = certificateId,
+                MeteringPointType = matchingContract.MeteringPointType
+            }));
+        builder.AddSubscription(issueCertificateFailedConsumerEndpoint, RoutingSlipEvents.Faulted,
+            x => x.Send(new IssueCertificateFaulted
+            {
+                CertificateId = certificateId,
+                MeteringPointType = matchingContract.MeteringPointType
+            }));
+    }
+
+    private void AddActivity<T, TArguments>(RoutingSlipBuilder routingSlipBuilder, TArguments arguments)
```

Comment:
What is an Activity? 
Do we have some documentation on this? It feels like a really have to understand wallet to understand this :)